torch/distributed/_tensor/debug/visualize_sharding.py

```diff
@@ -0,0 +1,160 @@
+import torch
+from torch.distributed._tensor import DeviceMesh, Shard, Replicate, distribute_tensor
+import os
+
+def shard_or_replicate(curr_ranges, placements, num_splits, idx, grid, device_mesh):
+    if len(placements) == idx:
+        assert not isinstance(device_mesh, list)
+        for i in curr_ranges['row']:
+            for j in curr_ranges['col']:
+                grid[i][j] = f"Device:{str(device_mesh)}" if grid[i][j] == "" else grid[i][j] + f", Device:{str(device_mesh)}"
```

Comment:
I also think we should use concrete device instead of "Device", we should inspect from `device_mesh.device_type` instead